rfcs/rfc-create-ui-architecture.md

```diff
@@ -0,0 +1,146 @@
+# RFC for a "Create UI" Architecture To Support New Filing Types
+
+- Start Date: (2021-03-10)
+- Target Major Version: None
+- Zenhub Ticket #: https://github.com/bcgov/entity/issues/6733
+
+## Introduction
+
+Currently, the [Create UI web app](https://github.com/bcgov/bcrs-business-create-ui
+) allows a user to enter and file a Benefit Company (BEN) Incorporation Application (IA). This web app needs to be enhanced to support many other types of incorporation and continuation filings.
+
+The purpose of this RFC is to propose/communicate/discuss/plan the implementation of an architecture in this web app to support these other filings. The goals of this architecture are to:
+
+- speed up the integration of new types of filings
+- promote easy-to-understand code flows as they relate to different filing types
+- promote maximum reuse of existing architecture, components and other code
+- facilitate maintenance of increasingly-complex code
+- minimize large future refactors
+- apply best practices learned in the development of other BCROS web apps (and other apps in general)
+
+## Summary
+
+The requirement is for a way to capture business logic in the code, but have it easy to work with.
+
+The proposed solution is two-fold:
+
+1. centralize the differences of the various filing types in one place -- using a "resources" (aka config) object that contains data specific to each filing type
+
+2. update the code to function differently, where needed, according to each filing's configuration
+
+Note: This discussion excludes the use of a high-level rules engine such as [json-rules-engine](https://www.npmjs.com/package/json-rules-engine), which is deemed overly generic and likely to lead to more complexity, not less.
```

Comment:
I did not suggest using this rules engine, what I did suggest was looking at some of the syntax.
Always useful to learn from others.
Also if there is some common terming used, it would be good to leverage that.

Reply:
Note: let's look at the syntax for specifying rules/dates/etc and maybe do something similar.

Reply:
Updated.